core/src/main/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbe.java

```diff
@@ -76,25 +87,62 @@ protected boolean requiresRelease() {
         return true;
     }
 
-    private ProbeResult fromSCMLink(ProbeContext context, String scm) {
+    private ProbeResult fromSCMLink(ProbeContext context, String scm, String pluginName) {
         Matcher matcher = GH_PATTERN.matcher(scm);
         if (!matcher.find()) {
             if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug("{} is not respecting the SCM URL Template", scm);
+                LOGGER.debug("{} is not respecting the SCM URL Template.", scm);
             }
-            return ProbeResult.failure(key(), "SCM link doesn't match GitHub plugin repositories");
+            return ProbeResult.failure(key(), "SCM link doesn't match GitHub plugin repositories.");
         }
-
         try {
             context.getGitHub().getRepository(matcher.group("repo"));
-            return ProbeResult.success(key(), "The plugin SCM link is valid");
+            Optional<Path> file = searchPomFiles(context.getScmRepository(), pluginName);
+            String folderPath = file.isPresent() ? new File(file.toString()).getParentFile().getName() : "";
+            context.setScmFolderPath(folderPath);
+            return ProbeResult.success(key(), "The plugin SCM link is valid.");
         } catch (IOException ex) {
-            return ProbeResult.failure(key(), "The plugin SCM link is invalid");
+            return ProbeResult.failure(key(), "The plugin SCM link is invalid.");
         }
     }
 
     @Override
     public String[] getProbeResultRequirement() {
         return new String[]{UpdateCenterPluginPublicationProbe.KEY};
     }
+
+    /**
+     * Searches for Pom files in every directory available in the repository
+     *
+     * @param directory  in the scm
+     * @param pluginName the name of the plugin
+     * @return folderPath if it valid or return the scm itself
+     */
+    private Optional<Path> searchPomFiles(Path directory, String pluginName) {
+        Optional<Path> fileName = Optional.empty();
+        try (Stream<Path> paths = Files.find(directory, 2, (path, $) ->
```

Comment:
You are right about this. 

I discussed with Adrien in our last weekly meeting. And he said `2` is the ideal depth. He also added that we can go till 5 but it would be a little crazy number.

I was thinking maybe we should find the max depth of a directory and then set it accordingly.

Reply:
I think whatever the value is, it should be clear why we chose that value.

I'd go for either

- Use Integer.MAX_VALUE so we scan everything
- Use a fixed value that we consider the maximum supported/recommended value and we document it as such

Reply:
I went for Integer.MAX_VALUE

Reply:
The rationale looks good to me, I just think it needs to be explicit and clear in the source code.